src/core/dev_api/tensor_data_accessor.hpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/node.hpp"
+#include "openvino/runtime/tensor.hpp"
+
+namespace ov {
+/** @brief Interface for data accessor. */
+class OPENVINO_API ITensorAccessor {
```

Comment:
It not replace the ov::tensor it provide interface how get e.g like plugin own implementation.
But if the `MemoryAccesser` can be removed because of the use the `RemoteTensor` then can remove it.